compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/FirSpecificTypeResolverTransformer.kt

```diff
@@ -65,7 +79,7 @@ class FirSpecificTypeResolverTransformer(
     ): FirResolvedTypeRef {
         functionTypeRef.transformChildren(this, data)
         session.lookupTracker?.recordTypeLookup(functionTypeRef, data.scope.scopeOwnerLookupNames, currentFile?.source)
-        val resolvedType = typeResolver.resolveType(functionTypeRef, data, areBareTypesAllowed).takeIfAcceptable()
+        val resolvedType = typeResolver.resolveType(functionTypeRef, data, areBareTypesAllowed, isOperandOfIsOperator = false).takeIfAcceptable()
```

Comment:
Despite of the fact `FirFunctionTypeRef` cannot refer to enum entry, I'd suggest to keep correct value of `isOperandOfIsOperator` here (to keep the code clearness).

Reply:
Sounds good. Thanks!